modules/exoplanets/exoplanetsmodule_lua.inl

```diff
@@ -56,120 +56,125 @@ constexpr const char* NoDataTextureFile =
     // 2. go to that location in the data file
     // 3. read sizeof(exoplanet) bytes into an exoplanet object.
-    Exoplanet p;
+    ExoplanetDataEntry p;
     std::string line;
-    bool found = false;
-
-    std::vector<Exoplanet> planetSystem;
-    std::vector<std::string> planetNames;
-
     while (getline(lut, line)) {
         std::istringstream ss(line);
         std::string name;
         getline(ss, name, ',');
 
-        if (name.compare(0, name.length() - 2, starNameSpeck) == 0) {
+        if (name.compare(0, name.length() - 2, starName) == 0) {
             std::string location_s;
             getline(ss, location_s);
             long location = std::stol(location_s.c_str());
 
             data.seekg(location);
-            data.read(reinterpret_cast<char*>(&p), sizeof(Exoplanet));
+            data.read(reinterpret_cast<char*>(&p), sizeof(ExoplanetDataEntry));
 
             sanitizeNameString(name);
-            planetNames.push_back(name);
-            planetSystem.push_back(p);
-            found = true;
 
             if (!hasSufficientData(p)) {
-                LERROR(fmt::format(
-                    "Insufficient data available for exoplanet system: '{}'",
-                    starName
-                ));
-                return;
+                LWARNING(fmt::format("Insufficient data for exoplanet: '{}'", name));
+                continue;
+            }
+
+            system.planetNames.push_back(name);
+            system.planetsData.push_back(p);
+
+            // Star data - Should not vary between planets, but one data entry might
+            // lack data for the host star while another does not. So for every planet,
+            // update star data if needed
+            const glm::vec3 pos{ p.positionX, p.positionY, p.positionZ };
+            if (system.starData.position != pos && isValidPosition(pos)) {
+                system.starData.position = pos;
+            }
+            if (system.starData.bvColorIndex != p.bmv && !std::isnan(p.bmv)) {
+                system.starData.bvColorIndex = p.bmv;
+            }
+            if (system.starData.radius != p.rStar && !std::isnan(p.rStar)) {
+                system.starData.radius = p.rStar;
             }
         }
     }
 
-    data.close();
-    lut.close();
+    system.starName = starName;
+    return std::move(system);
+}
+
+void createExoplanetSystem(const std::string& starName) {
+    const std::string starIdentifier = createIdentifier(starName);
+
+    std::string sanitizedStarName = starName;
+    sanitizeNameString(sanitizedStarName);
+
+    const std::string guiPath = ExoplanetsGuiPath + sanitizedStarName;
 
-    if (!found) {
-        LERROR(fmt::format("No star with the provided name was found: '{}'", starName));
+    SceneGraphNode* existingStarNode = sceneGraphNode(starIdentifier);
+    if (existingStarNode) {
+        LERROR(fmt::format(
+            "Adding of exoplanet system '{}' failed. The system has already been added.",
```

Comment:
We leave out the `.` at the end of warning/error messages

Reply:
Even multiple sentence ones? Okay!